Generate serializers for FontTaggedSettings.h

beidson opened the pull request:
#### 5856cad983b6efaf26dddec0982d681ef62c6210
<pre>
Generate serializers for FontTaggedSettings.h
<a href="https://bugs.webkit.org/show_bug.cgi?id=262138">https://bugs.webkit.org/show_bug.cgi?id=262138</a>
rdar://116079823

Reviewed by Alex Christensen.

* Source/WebCore/platform/graphics/FontTaggedSettings.h:
(WebCore::FontTaggedSetting&lt;T&gt;::encode const): Deleted.
(WebCore::FontTaggedSetting&lt;T&gt;::decode): Deleted.
(WebCore::FontTaggedSettings&lt;T&gt;::encode const): Deleted.
(WebCore::FontTaggedSettings&lt;T&gt;::decode): Deleted.
* Source/WebKit/Shared/WebCoreArgumentCoders.serialization.in:

Canonical link: <a href="https://commits.webkit.org/268488@main">https://commits.webkit.org/268488@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/9cd059db3dc06c069b064abd436ef3c602e300c6

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/19831 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/20255 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/20873 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/21723 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/18523 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/20066 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/23513 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/20402 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/20066 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/20051 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/20038 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/17247 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/22577 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/17210 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/18042 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/24322 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/18286 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/18216 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/22305 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/18806 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/17/builds/15936 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/17976 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/4744 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/22324 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/18631 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

bls1999 committed:
New Test (268092@main): [ Release ] WKWebExtensionAPITabs.RemovedEvent is flaky (sometimes wrong tab ID)
rdar://115809590
https://bugs.webkit.org/show_bug.cgi?id=261848

Unreviewed test gardening.

Disabling flaky API test.

* Tools/TestWebKitAPI/Tests/WebKitCocoa/WKWebExtensionAPITabs.mm:
(TestWebKitAPI::TEST):

Canonical link: https://commits.webkit.org/268482@main

bls1999 committed:
[Gardening]: REGRESSION(267193@main): [ Debug ] TestWebKitAPI.WebKit2.CrashGPUProcessWhileCapturingAndCalling is a near-constant failure.
rdar://116083315
https://bugs.webkit.org/show_bug.cgi?id=262145

Unreviewed test gardening.

Disabling consistently failing API test.

* Tools/TestWebKitAPI/Tests/WebKit/GetUserMedia.mm:

Canonical link: https://commits.webkit.org/268483@main

hortont424 committed:
Rename CGDisplayList and friends to their real names
https://bugs.webkit.org/show_bug.cgi?id=261938
rdar://115880900

Reviewed by Wenson Hsieh.

Rename and upstream various dynamic-content-scaling-related classes.
Also, remove a very noisy bit of logging, and a setting that is no longer needed.

* Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/platform/graphics/ImageBuffer.h:
* Source/WebCore/platform/graphics/cocoa/WebCoreCALayerExtras.h:
* Source/WebCore/platform/graphics/cocoa/WebCoreCALayerExtras.mm:
(-[CALayer _web_clearContents]):
(-[CALayer _web_clearDynamicContentScalingDisplayListIfNeeded]):
(-[CALayer _web_clearCGDisplayListIfNeeded]): Deleted.
* Source/WebCore/platform/graphics/re/DynamicContentScalingTypes.h: Copied from Source/WebCore/platform/graphics/cocoa/WebCoreCALayerExtras.h.
* Source/WebKit/Shared/RemoteLayerTree/DynamicContentScalingDisplayList.cpp: Renamed from Source/WebKit/Shared/RemoteLayerTree/CGDisplayList.cpp.
(WebKit::DynamicContentScalingDisplayList::encode):
(WebKit::DynamicContentScalingDisplayList::decode):
* Source/WebKit/Shared/RemoteLayerTree/DynamicContentScalingDisplayList.h: Renamed from Source/WebKit/Shared/RemoteLayerTree/CGDisplayList.h.
(WebKit::DynamicContentScalingDisplayList::DynamicContentScalingDisplayList):
(WebKit::DynamicContentScalingDisplayList::buffer const):
(WebKit::DynamicContentScalingDisplayList::takeSurfaces):
* Source/WebKit/Shared/RemoteLayerTree/DynamicContentScalingImageBufferBackend.h: Renamed from Source/WebKit/Shared/RemoteLayerTree/CGDisplayListImageBufferBackend.h.
* Source/WebKit/Shared/RemoteLayerTree/DynamicContentScalingImageBufferBackend.mm: Renamed from Source/WebKit/Shared/RemoteLayerTree/CGDisplayListImageBufferBackend.mm.
(WTF::CFTypeTrait<CAMachPortRef>::typeID):
(WebKit::makeContextOptions):
(WebKit::GraphicsContextDynamicContentScaling::GraphicsContextDynamicContentScaling):
(WebKit::DynamicContentScalingImageBufferBackend::calculateMemoryCost):
(WebKit::DynamicContentScalingImageBufferBackend::create):
(WebKit::DynamicContentScalingImageBufferBackend::DynamicContentScalingImageBufferBackend):
(WebKit::m_renderingMode):
(WebKit::DynamicContentScalingImageBufferBackend::createBackendHandle const):
(WebKit::DynamicContentScalingImageBufferBackend::context):
(WebKit::DynamicContentScalingImageBufferBackend::backendSize const):
(WebKit::DynamicContentScalingImageBufferBackend::bytesPerRow const):
(WebKit::DynamicContentScalingImageBufferBackend::releaseGraphicsContext):
(WebKit::DynamicContentScalingImageBufferBackend::copyNativeImage):
(WebKit::DynamicContentScalingImageBufferBackend::createNativeImageReference):
(WebKit::DynamicContentScalingImageBufferBackend::getPixelBuffer):
(WebKit::DynamicContentScalingImageBufferBackend::putPixelBuffer):
(WebKit::DynamicContentScalingImageBufferBackend::debugDescription const):
(WebKit::DynamicContentScalingAcceleratedImageBufferBackend::create):
(WebKit::DynamicContentScalingAcceleratedImageBufferBackend::DynamicContentScalingAcceleratedImageBufferBackend):
* Source/WebKit/Shared/RemoteLayerTree/RemoteLayerBackingStore.h:
* Source/WebKit/Shared/RemoteLayerTree/RemoteLayerBackingStore.mm:
(WebKit::RemoteLayerBackingStore::clearBackingStore):
(WebKit::hasValue):
(WebKit::RemoteLayerBackingStore::encode const):
(WebKit::RemoteLayerBackingStoreProperties::decode):
(WebKit::RemoteLayerBackingStore::swapToValidFrontBuffer):
(WebKit::RemoteLayerBackingStore::supportsPartialRepaint const):
(WebKit::RemoteLayerBackingStore::ensureFrontBuffer):
(WebKit::RemoteLayerBackingStore::paintContents):
(WebKit::RemoteLayerBackingStore::drawInContext):
(WebKit::RemoteLayerBackingStoreProperties::layerContentsBufferFromBackendHandle):
(WebKit::RemoteLayerBackingStoreProperties::applyBackingStoreToLayer):
* Source/WebKit/Shared/RemoteLayerTree/RemoteLayerTreePropertyApplier.mm:
(WebKit::RemoteLayerTreePropertyApplier::applyPropertiesToLayer):
* Source/WebKit/SourcesCocoa.txt:
* Source/WebKit/UIProcess/RemoteLayerTree/RemoteLayerTreeHost.h:
* Source/WebKit/UIProcess/RemoteLayerTree/RemoteLayerTreeHost.mm:
(WebKit::RemoteLayerTreeHost::replayDynamicContentScalingDisplayListsIntoBackingStore const):
(WebKit::RemoteLayerTreeHost::replayCGDisplayListsIntoBackingStore const): Deleted.
* Source/WebKit/UIProcess/RemoteLayerTree/cocoa/RemoteLayerTreeLayers.h:
* Source/WebKit/UIProcess/RemoteLayerTree/cocoa/RemoteLayerTreeLayers.mm:
(-[WKCompositingLayer _setWKContents:withDisplayList:replayForTesting:]):
(-[WKCompositingLayer drawInContext:]):
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:
* Source/WebKit/WebProcess/GPU/graphics/ImageBufferBackendHandle.h:
* Source/WebKit/WebProcess/WebPage/RemoteLayerTree/PlatformCALayerRemote.h:
* Source/WebKit/WebProcess/WebPage/RemoteLayerTree/PlatformCALayerRemote.mm:
(WebKit::PlatformCALayerRemote::shouldIncludeDisplayListInBackingStore const):
(WebKit::PlatformCALayerRemote::updateBackingStore):
* Source/WebKit/WebProcess/WebPage/RemoteLayerTree/RemoteLayerTreeContext.h:
(WebKit::RemoteLayerTreeContext::useDynamicContentScalingDisplayListsForDOMRendering const):
(WebKit::RemoteLayerTreeContext::setUseDynamicContentScalingDisplayListsForDOMRendering):
(WebKit::RemoteLayerTreeContext::useCGDisplayListsForDOMRendering const): Deleted.
(WebKit::RemoteLayerTreeContext::setUseCGDisplayListsForDOMRendering): Deleted.
(WebKit::RemoteLayerTreeContext::useCGDisplayListImageCache const): Deleted.
(WebKit::RemoteLayerTreeContext::setUseCGDisplayListImageCache): Deleted.
* Source/WebKit/WebProcess/WebPage/RemoteLayerTree/RemoteLayerTreeDrawingArea.mm:
(WebKit::RemoteLayerTreeDrawingArea::updatePreferences):

Canonical link: https://commits.webkit.org/268484@main

achristensen07 commented on Source/WebKit/Shared/WebCoreArgumentCoders.serialization.in:
```diff
@@ -6144,3 +6144,27 @@ struct WebCore::VideoConfiguration {
+    Vector<WebCore::FloatFontTaggedSetting> m_list;
+};
+
+header: <WebCore/FontTaggedSettings.h>
```
This is only needed once.

whsieh committed:
Move off of UIKit SPI: UIWebFormAccessory and UIWebFormAccessoryDelegate
https://bugs.webkit.org/show_bug.cgi?id=262121

Reviewed by Tim Horton.

Stop using this SPI class and delegate protocol; instead, introduce `WKFormAccessoryView` and
`WKFormAccessoryViewDelegate`, which are (nearly) drop-in replacements for `UIWebFormAccessory`.
The only functionality on `UIWebFormAccessory` which we drop in our new WebKit class is support for
showing a "Clear" button (`-setClearVisible:`), which was only necessary when using the "wheels of
time" UI in the keyboard on iPhone; this hasn't been necessary since the form control refresh,
which moved date pickers into a context menu presentation (and more recently, into popover UI).

* Source/WebKit/Platform/spi/ios/UIKitSPI.h:

Remove now-unused SPI declarations.

* Source/WebKit/SourcesCocoa.txt:
* Source/WebKit/UIProcess/ios/WKContentViewInteraction.h:
* Source/WebKit/UIProcess/ios/WKContentViewInteraction.mm:
(-[WKFormInputSession accessoryViewCustomButtonTitle]):
(-[WKContentView accessoryDone]):
(-[WKContentView accessoryViewDone:]):
(-[WKContentView accessoryTab:]):
(-[WKContentView accessoryView:tabInDirection:]):

Instead of using the existing `-accessoryTab:` delegate that takes a `BOOL`, we can instead use an
`enum class` to represent the direction (next or previous) for slightly better clarity.

(-[WKContentView accessoryViewAutoFill:]):
(-[WKContentView formAccessoryView]):
(-[WKContentView _updateAccessory]):
(-[WKContentView accessoryClear]): Deleted.
(-[WKContentView accessoryAutoFill]): Deleted.

Make `_formAccessoryView` a `WKFormAccessoryView` instead of a `UIWebFormAccessory`, and conform to
`WKFormAccessoryViewDelegate` instead of `UIWebFormAccessoryDelegate`.

* Source/WebKit/UIProcess/ios/forms/WKFormAccessoryView.h: Added.
* Source/WebKit/UIProcess/ios/forms/WKFormAccessoryView.mm: Added.
(WebKit::doneButtonFont):
(WebKit::singleLineAutoFillButtonFont):
(WebKit::multipleLineAutoFillButtonFont):
(WebKit::upArrow):
(WebKit::downArrow):
(WebKit::createToolbarWithItems):
(-[WKFormAccessoryView _initForUniversalControlBar:]):
(-[WKFormAccessoryView initWithInputAssistantItem:delegate:]):
(-[WKFormAccessoryView traitCollectionDidChange:]):
(-[WKFormAccessoryView _updateFrame]):
(-[WKFormAccessoryView layoutSubviews]):
(-[WKFormAccessoryView _done]):
(-[WKFormAccessoryView _previousTapped]):
(-[WKFormAccessoryView _nextTapped]):
(-[WKFormAccessoryView _autoFill]):
(-[WKFormAccessoryView _autoFillButton]):
(-[WKFormAccessoryView _refreshAutofillPresentation]):
(-[WKFormAccessoryView _applyTreatmentToAutoFillLabel]):
(-[WKFormAccessoryView hideAutoFillButton]):
(-[WKFormAccessoryView showAutoFillButtonWithTitle:]):
(-[WKFormAccessoryView autoFillButtonItem]):
(-[WKFormAccessoryView setNextPreviousItemsVisible:]):
(-[WKFormAccessoryView setNextEnabled:]):
(-[WKFormAccessoryView isNextEnabled]):
(-[WKFormAccessoryView setPreviousEnabled:]):
(-[WKFormAccessoryView isPreviousEnabled]):
(-[WKFormAccessoryView contentRatio]):
* Source/WebKit/UIProcess/mac/DisplayCaptureSessionManager.mm:
(WebKit::DisplayCaptureSessionManager::promptForGetDisplayMedia):

Unrelated unified source build fix.

* Source/WebKit/WebKit.xcodeproj/project.pbxproj:
* Tools/TestWebKitAPI/Tests/ios/KeyboardInputTestsIOS.mm:
(TestWebKitAPI::TEST):
* Tools/TestWebKitAPI/ios/UIKitSPI.h:

Canonical link: https://commits.webkit.org/268485@main

bls1999 committed:
[Gardening]: REGRESSION(268408@main): [ macOS ] TestWebKitAPI.WebPushDPushNotificationEventTest.Basic is constantly timing out.
rdar://116087660
https://bugs.webkit.org/show_bug.cgi?id=262150

Unreviewed test gardening.

* Tools/TestWebKitAPI/Tests/WebKitCocoa/WebPushDaemon.mm:

Canonical link: https://commits.webkit.org/268486@main

beidson committed:
Replace CapabilityValueOrRange with CapabilityRange
https://bugs.webkit.org/show_bug.cgi?id=262099
rdar://116038461

Reviewed by Eric Carlson.

- Remove basically unused "value()" feature
- Do nothing by store ranges internall
- Update all users of the class to be nicer
- Generate encode/decode, with validated min/max

* Source/WebCore/Modules/mediastream/MediaTrackCapabilities.cpp:
(WebCore::capabilityDoubleRange):
(WebCore::capabilityIntRange):
* Source/WebCore/platform/mediastream/RealtimeMediaSource.cpp:
(WebCore::RealtimeMediaSource::supportsSizeFrameRateAndZoom):
(WebCore::RealtimeMediaSource::fitnessDistance):
(WebCore::RealtimeMediaSource::applyConstraint):
(WebCore::RealtimeMediaSource::extractVideoFrameSizeConstraints):
(WebCore::RealtimeMediaSource::applyConstraints):
* Source/WebCore/platform/mediastream/RealtimeMediaSourceCapabilities.h:
(WebCore::CapabilityValueOrRange::CapabilityValueOrRange):
(WebCore::CapabilityValueOrRange::rangeMin const):
(WebCore::CapabilityValueOrRange::rangeMax const):
(WebCore::CapabilityValueOrRange::value const):
(WebCore::CapabilityValueOrRange::encode const): Deleted.
(WebCore::CapabilityValueOrRange::decode): Deleted.
* Source/WebKit/Shared/WebCoreArgumentCoders.serialization.in:

Canonical link: https://commits.webkit.org/268487@main

beidson committed:
Generate serializers for FontTaggedSettings.h
https://bugs.webkit.org/show_bug.cgi?id=262138
rdar://116079823

Reviewed by Alex Christensen.

* Source/WebCore/platform/graphics/FontTaggedSettings.h:
(WebCore::FontTaggedSetting<T>::encode const): Deleted.
(WebCore::FontTaggedSetting<T>::decode): Deleted.
(WebCore::FontTaggedSettings<T>::encode const): Deleted.
(WebCore::FontTaggedSettings<T>::decode): Deleted.
* Source/WebKit/Shared/WebCoreArgumentCoders.serialization.in:

Canonical link: https://commits.webkit.org/268488@main
